.idea/readme.md

```diff
@@ -30,6 +30,17 @@ _**PyCharm’s remote debugger requires PyCharm Professional._
 
 <br />
 
+<h1>Setting up Python Code Completion</h1>
+
+In Pycharm or Rider, after installing [`lean-cli`](https://github.com/QuantConnect/lean-cli) or [`quantconnect-stubs`](https://github.com/QuantConnect/quantconnect-stubs-generator),
+
+1. Open settings
+2. Click on "Python Console" under the "Console" subsection
+3. For the interpreter of your choice, add `import quantconnect-stubs` under "Starting script"
+
+After clicking "Apply" or "Ok", all .py files opened using these project settings will implicitly import the stubs module. 
+It is recommended that a virtual environment is used for your Lean project so that these stubs will only ever load when developing in Lean.
```

Comment:
This doesn't do what it states it does for me. I followed these steps locally and all it added was an error every time I open a new Python console tab because `import`ing a name containing a hyphen is a syntax error (autocompletion in .py files doesn't seem to have changed). Do you get the same error if you open the Python console in the IDE after applying this change (should be in the bottom left)?

Reply:
You are correct -- I should have imported the individual imports as:
```from QuantConnect import *
from QuantConnect.Parameters import *
from QuantConnect.Benchmarks import *
from QuantConnect.Brokerages import *
from QuantConnect.Util import *
from QuantConnect.Interfaces import *
from QuantConnect.Algorithm import *
from QuantConnect.Algorithm.Framework import *
from QuantConnect.Algorithm.Framework.Selection import *
from QuantConnect.Algorithm.Framework.Alphas import *
from QuantConnect.Algorithm.Framework.Portfolio``` ...
- I'll change the recommended imports to paste to a link to the part of the ReadMe which lists these out.

For the installation source, lean-cli being installed would imply quantconnect-stubs to also be given that `install_requires` has the stubs as a dependency -- I figured I'd mention either should satisfy this. 

Lastly, for the error, I was getting it too found a non-elegant way to fix it. In my case, looking into my clr module, it had been replaced by stub files (i.e., clr only contained __init__.pyi). Having it there was actually preventing pythonnet from being installed too -- in my case, `rm -r (...your dir)/site-packages/clr` and then installing pythonnet allowed for importing as expected.

Reply:
Note that even after the fix, it may be that `clr` needs to be imported first:
```
import clr
import QuantConnect
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/home/ajst/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/203.7148.72/plugins/python/helpers/pydev/_pydev_bundle/pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "/home/ajst/miniconda3/envs/Lean/lib/python3.6/site-packages/QuantConnect/__init__.py", line 30, in <module>
    AddReference("QuantConnect")
System.IO.FileNotFoundException: Unable to find assembly 'QuantConnect'.
  at Python.Runtime.CLRModule.AddReference (System.String name) [0x00057]
...

```
Which is an error thrown after successfully importing + utilizing `clr`.

vs

```
import QuantConnect
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/home/ajst/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/203.7148.72/plugins/python/helpers/pydev/_pydev_bundle/pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
  File "/home/ajst/miniconda3/envs/Lean/lib/python3.6/site-packages/QuantConnect/__init__.py", line 29, in <module>
    from clr import AddReference
  File "/home/ajst/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/203.7148.72/plugins/python/helpers/pydev/_pydev_bundle/pydev_import_hook.py", line 21, in do_import
    module = self._system_import(name, *args, **kwargs)
ModuleNotFoundError: No module named 'clr'
```

Given that this seems to be a non-trivial issue regarding dll linking and packages external to QuantConnect/Lean, I will convert this PR into a draft for the time being.